ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -282,15 +282,12 @@ private ValidReaderWriteIdList getValidCleanerWriteIdList(CompactionInfo ci, Tab
     assert rsp != null && rsp.getTblValidWriteIdsSize() == 1;
     ValidReaderWriteIdList validWriteIdList =
         TxnCommonUtils.createValidReaderWriteIdList(rsp.getTblValidWriteIds().get(0));
-    boolean delayedCleanupEnabled = conf.getBoolVar(HIVE_COMPACTOR_DELAYED_CLEANUP_ENABLED);
-    if (delayedCleanupEnabled) {
-      /*
-       * If delayed cleanup enabled, we need to filter the obsoletes dir list, to only remove directories that were made obsolete by this compaction
-       * If we have a higher retentionTime it is possible for a second compaction to run on the same partition. Cleaning up the first compaction
-       * should not touch the newer obsolete directories to not to violate the retentionTime for those.
-       */
-      validWriteIdList = validWriteIdList.updateHighWatermark(ci.highestWriteId);
-    }
+    /*
+     * We need to filter the obsoletes dir list, to only remove directories that were made obsolete by this compaction
+     * If we have a higher retentionTime it is possible for a second compaction to run on the same partition. Cleaning up the first compaction
+     * should not touch the newer obsolete directories to not to violate the retentionTime for those.
+     */
+    validWriteIdList = validWriteIdList.updateHighWatermark(ci.highestWriteId);
```

Comment:
How do we know that ci.highestWriteId's txn <= the min open txn the cleaner uses, if MIN_HISTORY_LEVEL is still used?

Reply:
not sure I got the question. but highestWriteId is recorded at the time when the compaction txn starts, so it records all open txns that have to be ignored.

Reply:
Yes, ci.highestWriteId = the highest write id that was compacted.
So if we have this after compaction:
delta_1_1
delta_2_2
delta_3_3
base_3
ci.highestWriteId=3, so the cleaner will remove (assuming MIN_HISTORY_LEVEL is still being used) : 
delta_1_1
delta_2_2
delta_3_3
But how do we know those can be removed?

Reply:
validWriteIdList besides the updated hwm has also an exceptions list, that would show if there are any open txns in that range. What we are doing here is just lowering the hwm so that cleaner won't remove more than this compaction was responsible for.

Reply:
And we're 100% sure that we're lowering it and not raising it? Maybe we could include some sort of assertion that ci.highestWriteId <= previous high watermark?

Reply:
sure, that could be done